src/zerocash/Address.cpp

```diff
@@ -11,26 +11,14 @@
  * @copyright  MIT license (see LICENSE file)
  *****************************************************************************/
 
-#include <cryptopp/osrng.h>
-using CryptoPP::AutoSeededRandomPool;
-
-#include <cryptopp/eccrypto.h>
-using CryptoPP::ECP;
-using CryptoPP::ECIES;
-
-#include <cryptopp/oids.h>
-namespace ASN1 = CryptoPP::ASN1;
-
-#include <cryptopp/filters.h>
-using CryptoPP::StringSink;
-using CryptoPP::StringStore;
+#include "zcash/NoteEncryption.hpp"
 
 #include "Zerocash.h"
 #include "Address.h"
 
 namespace libzerocash {
 
-PrivateAddress::PrivateAddress(const std::vector<unsigned char> a_sk, const std::string sk_enc) {
+PrivateAddress::PrivateAddress(const uint256 &a_sk, const uint256 &sk_enc) {
```

Comment:
I'm not convinced this is the correct type. The most obvious representation of these keys is as byte strings (with the MS bit first in each byte as per the spec). ``uint256`` introduces endianness issues into this API that should be the responsibility of the NoteEncryption itself, not its users. I think I'd prefer something like:
```
typedef unsigned char SpendingKey[32];
typedef unsigned char NoteDecryptionKey[32];
```
(There are two changes here, the typedef and the change to use an unsigned char array. The former is probably more important.)